pkg/core/klinebuilder.go

```diff
@@ -0,0 +1,190 @@
+package core
+
+import (
+	"context"
+	"sync"
+	"time"
+
+	"github.com/c9s/bbgo/pkg/fixedpoint"
+	"github.com/c9s/bbgo/pkg/types"
+	"github.com/sirupsen/logrus"
+)
+
+// KLineBuilder is a kline builder for a symbol
+// It will accumulate trades and build klines
+//
+//go:generate callbackgen -type kLineBuilder
+type kLineBuilder struct {
+	symbol      string
+	minInterval types.Interval
+	intervals   map[types.Interval]struct{}
+	// klinesMap is a map from interval to accumulated kline of that interval
+	klinesMap map[types.Interval]*types.KLine
+
+	kLineCallbacks       []func(kline types.KLine)
+	kLineClosedCallbacks []func(kline types.KLine)
+	logger               *logrus.Entry
+
+	// the following fields should be protected by lock when being updated
+	mu                                                 sync.Mutex
+	open, high, low, close, volume, quoteVolume, price fixedpoint.Value
```

Comment:
quote volume needs to be a float64 number, it's a huge number, especially in the BTCUSDT market

Reply:
How about volume?

Reply:
Oh, I know what's wrong here.
`quoteVolume` is the quote volume of the quote quantity of the last trade.
So, it is ok to use `fixedpoint.Value` here.

And due to the definition of `types.KLine`, I really can't use `float64` for the accumulated quote volume for a long interval.

Reply:
Or we'd have to update all codes where the `types.KLine` is applied.